External/FEXCore/Source/Interface/Core/ArchHelpers/Arm64Emitter.cpp

```diff
@@ -192,6 +192,23 @@ void Arm64Emitter::LoadConstant(ARMEmitter::Size s, ARMEmitter::Register Reg, ui
     return;
   }
 
+  if ((Constant >> 32) == 0) {
+    // If the upper 32-bits is all zero, we can now switch to a 32-bit move.
+    s = ARMEmitter::Size::i32Bit;
+    Is64Bit = false;
+    Segments = 2;
+  }
+
+  // If this can be loaded with a mov bitmask.
+  const auto IsImm = vixl::aarch64::Assembler::IsImmLogical(Constant, RegSizeInBits(s));
+  if (IsImm) {
+    orr(s, Reg, ARMEmitter::Reg::zr, Constant);
```

Comment:
I assume arm64 could do "x | 0x80808080" in one op but FEX after this PR will generate two, and we would need an algebraic optimizer to fuse that?

Reply:
There is some code in the ConstProp pass to determine if it can determine merge logical imm encodings in to an InlineConstant op. So theoretically it should already do that. This just missed the simple case of something like `mov rax, 0x80808080`

Reply:
Got it.